src/fs_fd.ts

```diff
@@ -78,6 +78,7 @@ export class OpenFile extends Fd {
     iovs: Array<wasi.Ciovec>
   ): { ret: number; nwritten: number } {
     let nwritten = 0;
+    if (this.file.readonly) return { ret: wasi.ERRNO_ROFS, nwritten };
```

Comment:
`ERRNO_ROFS` is for when the entire filesystem is read-only. Linux returns EBADF which corresponds to `ERRNO_BADF` in WASI.

Reply:
Aye, I switched them out. I didn't find some lengthier explanations for each `ERRNO`, so [I just picked one](https://github.com/bytecodealliance/wasi/blob/main/src/lib_generated.rs) that said "read only". Now that you've said that, I quickly found explanations [from The Open Group](https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html):

> [EBADF]
    Bad file descriptor. A file descriptor argument is out of range, refers to no open file, or a read (write) request is made to a file that is only open for writing (reading).